bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/OmnilinkHandlerFactory.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.omnilink.internal;
+
+import static org.openhab.binding.omnilink.internal.OmnilinkBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.omnilink.internal.handler.AudioSourceHandler;
+import org.openhab.binding.omnilink.internal.handler.AudioZoneHandler;
+import org.openhab.binding.omnilink.internal.handler.ButtonHandler;
+import org.openhab.binding.omnilink.internal.handler.ConsoleHandler;
+import org.openhab.binding.omnilink.internal.handler.HumiditySensorHandler;
+import org.openhab.binding.omnilink.internal.handler.LockHandler;
+import org.openhab.binding.omnilink.internal.handler.LuminaAreaHandler;
+import org.openhab.binding.omnilink.internal.handler.OmniAreaHandler;
+import org.openhab.binding.omnilink.internal.handler.TempSensorHandler;
+import org.openhab.binding.omnilink.internal.handler.ThermostatHandler;
+import org.openhab.binding.omnilink.internal.handler.UnitHandler;
+import org.openhab.binding.omnilink.internal.handler.ZoneHandler;
+import org.openhab.binding.omnilink.internal.handler.units.DimmableUnitHandler;
+import org.openhab.binding.omnilink.internal.handler.units.FlagHandler;
+import org.openhab.binding.omnilink.internal.handler.units.OutputHandler;
+import org.openhab.binding.omnilink.internal.handler.units.UpbRoomHandler;
+import org.openhab.binding.omnilink.internal.handler.units.dimmable.UpbUnitHandler;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link OmnilinkHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Craig Hamilton - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.omnilink")
+public class OmnilinkHandlerFactory extends BaseThingHandlerFactory {
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
```

Comment:
@ecdye where is the controller/bridge thing returned here ?  I'm not sure how this is working for you as you cannot create a bridge/controller  thing.